src/core/shader.js

```diff
@@ -0,0 +1,156 @@
+var debug = require('../utils/debug');
+var schema = require('./schema');
+var error = debug('core:register-shader:error');
+
+var processSchema = schema.process;
+var shaders = module.exports.shaders = {};  // Keep track of registered shaders.
```

Comment:
OK I see why we want `shaderList` now.

Can we call it `shaderNames` to distinguish it from actual shaders, and then comment why we need it